wsireg/utils/im_utils.py

```diff
@@ -84,6 +84,39 @@ def zarr_get_base_pyr_layer(zarr_store):
     return zarr_im
 
 
+def read_array(img, preprocessing, force_rgb=None):
+    """Read np.array, zarr.Array, or dask.array image into memory."""
+    is_interleaved = guess_rgb(img.shape)
+    is_rgb = is_interleaved if not force_rgb else force_rgb
+
+    if is_rgb:
+        if preprocessing is not None:
+            image = np.asarray(grayscale(img, is_interleaved=is_interleaved))
+            image = sitk.GetImageFromArray(image)
+        else:
+            image = np.asarray(img)
+            if not is_interleaved:
+                image = np.rollaxis(image, 0, 3)
+            image = sitk.GetImageFromArray(image, isVector=True)
+
+    elif len(img.shape) == 2:
+        image = sitk.GetImageFromArray(np.asarray(img))
+
+    else:
+        if preprocessing is not None:
+            if (
+                preprocessing.get("ch_indices") is not None
+                and len(img.shape) > 2
+            ):
+                chs = tuple(preprocessing.get('ch_indices'))
+                img = np.squeeze(np.asarray(img))
```

Comment:
I like the simplicity of `read_array`. One thing though, calling `np.asarray` here defeats one purpose of using `ch_indices`. If the array here were lazily-loaded, it would read into memory. One advantage of selecting channels at preprocessing is that you can avoid reading all planes into memory and only selecting the one needed for registration.

Reply:
Ah, good catch – I agree completely. I guess for the type of lazy indexing that is desired, requiring in-memory objects to be dask.arrays is totally reasonable.